File: output/ifc4x3_dev_73740fe4/entity/IfcLibraryReference.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


import java.util.Set;

public class IfcLibraryReference extends IfcExternalReference {

    public IfcText Description; // OPTIONAL
    public IfcLanguageId Language; // OPTIONAL
    public IfcLibraryInformation ReferencedLibrary; // OPTIONAL
    // INVERSE attributes:
    // LibraryRefForObjects : SET [0:?] OF IfcRelAssociatesLibrary FOR RelatingLibrary;
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcExternalReference/IfcLibraryReference/IfcLibraryReference.js
class IfcLibraryReference extends IfcExternalReference {
    constructor() {
        /** @type {IFCTEXT} */
        this.Description = null;
        /** @type {IFCLANGUAGEID} */
        this.Language = null;
        /** @type {IFCLIBRARYINFORMATION} */
        this.ReferencedLibrary = null;
    }

    // === INVERSE CLAUSES ===
    // LibraryRefForObjects : SET [0:?] OF IfcRelAssociatesLibrary FOR RelatingLibrary
}
